### PR TITLE
Make `sidecar/rev-a.toml` clippy-clean

### PR DESCRIPTION
**Pull request by mkeeter**
Clippy doesn't want us to construct needless references, and _really_ wants you to derive `Eq` if you're deriving `PartialEq`.

### CODE REVIEW

File: drv/vsc85xx/src/atom.rs [hunk 1]
```diff
@@ -8,9 +8,7 @@ use vsc7448_pac::phy;
 use vsc_err::VscError;
 
 /// Based on `vtss_atom_patch_suspend` in the SDK
-pub fn atom_patch_suspend<'a, 'b, P: PhyRw>(
-    phy: &'b mut Phy<'a, P>,
-) -> Result<(), VscError> {
+pub fn atom_patch_suspend<P: PhyRw>(phy: &mut Phy<P>) -> Result<(), VscError> {
```
Comment (cbiffle) [inline drv/vsc85xx/src/atom.rs:11]: I think you want `Phy<'_, P>` here, and probably also to set `deny(elided_lifetimes_in_paths)` at the crate root. I wish there was a way we could set this at the workspace level, but, there does not appear to be.

File: drv/vsc85xx/src/atom.rs [hunk 2]
```diff
@@ -23,9 +21,7 @@ pub fn atom_patch_suspend<'a, 'b, P: PhyRw>(
 }
 
 /// Based on `vtss_atom_patch_suspend` in the SDK
-pub fn atom_patch_resume<'a, 'b, P: PhyRw>(
-    phy: &'b mut Phy<'a, P>,
-) -> Result<(), VscError> {
+pub fn atom_patch_resume<P: PhyRw>(phy: &mut Phy<P>) -> Result<(), VscError> {
```
Comment (cbiffle) [inline drv/vsc85xx/src/atom.rs:24]: (same)

**Review by cbiffle — APPROVED** [review]
Please fix the two elided lifetime cases I've identified in the comments below, but otherwise looks good.

File: drv/vsc85xx/src/atom.rs (hunk 1 above)
Comment (mkeeter) [reply drv/vsc85xx/src/atom.rs]: I ended up turning it on as a Clippy lint (in `cargo xtask clippy`), then fixed it _everywhere_!

**Pull request merged**